Assets/Scripts/DraggableItem.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DraggableItem : MonoBehaviour
+{
+    private List<Transform> touchingItemSlots;
+    private Vector2 startPos;
+    private Transform myParent;
+
+    private void Awake()
+    {
+        startPos = transform.position;
+        touchingItemSlots = new List<Transform>();
+    }
+
+    public void PickUp()
+    {
+        transform.parent = myParent;
+        touchingItemSlots.Clear();
+        transform.localScale = new Vector3(6f, 6f, 6f);
+        gameObject.GetComponent<SpriteRenderer>().sortingOrder = 1;
+    }
+
+    public void Drop()
+    {
+        transform.localScale = new Vector3(5f, 5f, 5f);
+        gameObject.GetComponent<SpriteRenderer>().sortingOrder = 0;
+
+        Vector2 newPos;
+        if (touchingItemSlots.Count == 0)
+        {
+            transform.position = startPos;
+            transform.parent = myParent;
+            return;
+        }
+
+        var currentItemSlot = touchingItemSlots[0];
+        if(touchingItemSlots.Count == 1)
+        {
+            newPos = currentItemSlot.position;
+        }
+        else
+        {
+            var distance = Vector2.Distance(transform.position, touchingItemSlots[0].position);
+
+            foreach(Transform itemSlot in touchingItemSlots)
+            {
+                if(Vector2.Distance(transform.position, itemSlot.position) < distance)
+                {
+                    currentItemSlot = itemSlot;
+                    distance = Vector2.Distance(transform.position, itemSlot.position);
+                }
+            }
+            newPos = currentItemSlot.position;
+        }
```

Comment:
Deze class zorgt voor 2 dingen: 

1. maakt objecten draggable.
2. zorgt er voor dat objecten snappen in item slots.

Het zou beter zijn als deze class verantwoordelijk was voor alleen maar drag en droppen. 
Je zou een interface kunnen maken die werkt als 'Dropzone'. De specifieke dropzone classes die implementen dan deze interface. De drag en drop code kijkt of waar hij gedropt wordt een dropzone ligt, en dan voert hij simpelweg de `Dropzone.Drop()` functie uit. Door dit kan je gemakkelijk nieuwe type dropzones maken, namelijk: 

- Dropzones waar 1 object kan
- Dropzones waar ze meerdere naast elkaar gooien
- Dropzones waar de objecten samengeklikt worden.

En dan is het logischer wat elk script doet.